src/onemod/io/base.py

```diff
@@ -182,6 +242,32 @@ class Output(IO):
     """Stage output class."""
 
     items: dict[str, Data] = {}  # defined by stage class
+    _expected_names: set[str]
```

Comment:
For the output class, it would be created by the stage definition and be the same across all instances, so I was thinking that doing checks and serialization seemed unnecessary. But we can talk about it if you've changed how the `Input`/`Output` classes work!